bin/test-task.js

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env node
+"use strict";
+
+var util = require("util");
+
+var assert = require("assert"),
+    async = require("async"),
+    fs = require("fs"),
+    tmp = require("tmp"),
+    env = require("require-env");
+
+var worker = require("../lib/worker"),
+    upload = require("../lib/localUploader.js");
+
+var runWorker = function(task, callback) {
+  worker(0, task, upload, function(err) {
+    if (err) {
+      callback(err);
```

Comment:
I'm also partial to early returns, as they tend to avoid bugs like the one here: if an error is passed, `callback` will be called twice.